### PR TITLE
[mac] Adopt new save/open API

### PR DESCRIPTION
**Pull request by cmyr**
This implements the new save/open (#1068) API on macOS.

The implementation here is slightly different; Cocoa uses the idea
of 'blocks' (a c extension for callbacks) in most of their async API,
so we don't need to to manually defer the work.

### CODE REVIEW

File: druid-shell/src/platform/mac/window.rs [hunk 1]
```diff
@@ -933,26 +934,56 @@ impl WindowHandle {
...
+    pub fn save_as_sync(&mut self, _options: FileDialogOptions) -> Option<FileInfo> {
+        log::warn!("save_as_sync should no longer be called on mac!");
         None
     }
 
+    pub fn save_as(&mut self, options: FileDialogOptions) -> Option<FileDialogToken> {
+        let token = FileDialogToken::next();
+        let self_clone = self.clone();
+        unsafe {
+            let panel = dialog::build_panel(FileDialogType::Save, options);
```
Comment (jneem) [inline druid-shell/src/platform/mac/window.rs:971]: Looks like except for `FileDialogType::Save` this is identical to `open_file`?

**Review by jneem — APPROVED** [review]
Great! I have no way of testing this, but at least it *looks* good.

File: druid-shell/src/platform/mac/window.rs (hunk 1 above)
Comment (cmyr) [reply druid-shell/src/platform/mac/window.rs:971]: also different is the method we call on the handler inside the block. I could probably come up with some way to share this code more but I'm not sure it's worth the typing.

Comment (cmyr) [reply druid-shell/src/platform/mac/window.rs:971]: (okay fine I rewrote it to share an implementation 😛)

**Pull request merged**